cats/cats-core/src/main/java/com/netflix/spinnaker/cats/provider/DefaultProviderCache.java

```diff
@@ -145,7 +145,9 @@ public void putCacheResult(
         previousSet = new HashSet<>();
       }
       if (cacheResult.getCacheResults().containsKey(type)) {
-        cacheDataType(type, sourceAgentType, cacheResult.getCacheResults().get(type));
+        if (authoritativeTypes.contains(type)) {
```

Comment:
this will never populate relationship data when we are dealing with a full cache result from a caching agent (not the on demand refresh of a single entity)

I think we need to rethink this or we may be missing some of the relationship links for entities (especially for the redis cache implementation)

Reply:
IIUC, full cache result comes from https://github.com/spinnaker/clouddriver/blob/91515774dcfa43835efcdcfa993a0ed9efa1bffa/clouddriver-api/src/main/java/com/netflix/spinnaker/cats/agent/CachingAgent.java#L90

which later computes authoritative types:
https://github.com/spinnaker/clouddriver/blob/91515774dcfa43835efcdcfa993a0ed9efa1bffa/clouddriver-api/src/main/java/com/netflix/spinnaker/cats/agent/CachingAgent.java#L94-L99

then passed to `putCacheResult`/`addCacheResult`:
https://github.com/spinnaker/clouddriver/blob/91515774dcfa43835efcdcfa993a0ed9efa1bffa/clouddriver-api/src/main/java/com/netflix/spinnaker/cats/agent/CachingAgent.java#L138-L142

So we should still have agents populating the data, although now only authoritative data from these agents are used. Are there other sources that I missed?

Reply:
I did a quick look at `SqlProviderCache`, it seems it also cache only authoritative data types.  https://github.com/spinnaker/clouddriver/blob/53f9d33362c02b0bfe4b9ca86a1ea9e6cf9131b7/cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/SqlProviderCache.kt#L212-L220

Reply:
FWIW the `isPartialResult` doesn't seem to be used much (at least in the cloud providers that I have enabled currently anyway) and defaults to false so I'm not sure it is 100% the right thing to expect we rely on. In the majority of cases we'll be going through the putCacheResult for all cache updates

In the SQL cache there isn't a need to cache non-authoritative results because SQL gives you the ability to join

In the Redis implementation we store a pointer each direction (e.g. from cluster -> serverGroup and from serverGroup -> cluster) and without populating the non-authoritative we may be left with the inability to link from object to another